integtest/spec/all_books_change_detection_spec.rb

```diff
@@ -303,6 +322,16 @@ def chapter(index)
           )
           include_examples 'second build is noop'
         end
+        context 'even when there are noop changes source repo' do
```

Comment:
This doesn't actually reproduce the problem but it was what I tried first when I realized we didn't have it. i think it is worth having.